scripts/new-version-pr-generator.js

```diff
@@ -0,0 +1,180 @@
+      4. Update "shopify/polaris" by running "npx yarn upgrade @shopify/polaris@[release-version]"
+      5. Add the changed files to git by running "git add package.json yarn.lock"
+      6. Commit the changes with "git commit -m 'Update @shopify/polaris to [release-version]'
+      7. Push the changes with "git push origin update-polaris-v[release-version]"
+      8. Create a new pull request for the branch in "${repository}"`;
+
+      // Pull request template
+      const updateBody = `
+## Version ${releaseVersion} of @shopify/polaris just got published!
+
+See what’s new: https://github.com/Shopify/polaris-react/releases/tag/${releaseVersion}
+
+cc @kaelig @dfmcphee @amrocha @BPScott
+
+---
+
+<details>
+<summary>🚨 What to do if you see “Your tests failed on CircleCI”?</summary>
+
+If tests fail, you may have to troubleshoot the problem locally.
+
+1. Checkout the \`update-polaris-${releaseVersion}\` branch:
+    \`\`\`bash
+    dev cd ${repository}
+    git fetch
+    git checkout update-polaris-${releaseVersion}
+    dev up
+    \`\`\`
+1. Apply changes/fixes
+1. Commit
+1. Push your work to the \`update-polaris-${releaseVersion}\` branch:
+    \`\`\`bash
+    git push origin update-polaris-${releaseVersion}
+    \`\`\`
+1. Repeat until the build & tests go ✅
+1. 🎩
+1. :shipit:
+</details>
+`.trim();
+
+      const updatePostObject = {
+        title: `Update @shopify/polaris to ${releaseVersion} 🚀`,
+        body: updateBody,
+        head: `update-polaris-${releaseVersion}`,
+        base: baseBranch,
+      };
+
+      // Clone the repository to the repository directory
+      execSync(
+        `git clone --branch ${baseBranch} --single-branch https://${polarisBotToken}@github.com/Shopify/${repository}.git`,
+        {
+          cwd: sandbox,
+          stdio: 'inherit',
+        },
+      );
+
+      // Create a bunch of tasks
+      const commands = BulkExecute(
+        [
+          `git checkout -b update-polaris-${releaseVersion}`,
+          'npm config set @shopify:registry https://registry.npmjs.org',
+          `npx yarn@${YARN_VERSION} upgrade @shopify/polaris@${releaseVersion.replace(
+            'v',
+            '',
+          )} --no-progress --ignore-engines``git add package.json yarn.lock`,
+          `${shopifyPolarisBotGitOverride} git commit -m "Update @shopify/polaris to ${releaseVersion}" --author "${polarisBotName} <${polarisBotEmail}>" -m "${releaseVersion}" --allow-empty`,
+          `git push origin update-polaris-${releaseVersion}`,
+          `curl -d '${JSON.stringify(
+            updatePostObject,
+          )}' -X POST https://api.github.com/repos/shopify/${repository}/pulls?access_token=${polarisBotToken}`,
+        ],
+        {
+          cwd: repositoryDirectory,
+          stdio: 'inherit',
+        },
+      );
+
+      // Try and run update polaris using retry default settings
+      // 3 attempts maximum with a 10 second delay between each attempt
+      Retry(commands, failedUpdateMessage);
```

Comment:
This isn't doing what we want it to.

Right now, if the last request fails you retry every single one of these requests. What we want is to retry that individual request.